pkg/webhook/controlplaneexposure/ensurer_test.go

```diff
@@ -76,6 +76,34 @@ var _ = Describe("Ensurer", func() {
 			checkETCDMain(etcd)
 		})
 
+		It("should not modify existing elements of etcd-main statefulset which are 80Gi", func() {
+			var (
+				rOld    = resource.MustParse("80Gi")
+				oldEtcd = &druidv1alpha1.Etcd{
+					ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.ETCDMain},
+					Spec: druidv1alpha1.EtcdSpec{
+						StorageCapacity: &rOld,
+					},
+				}
+				rNew    = resource.MustParse("10Gi")
+				newEtcd = &druidv1alpha1.Etcd{
+					ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.ETCDMain},
+					Spec: druidv1alpha1.EtcdSpec{
+						StorageCapacity: &rNew,
+					},
+				}
+			)
+
+			// Create ensurer
+			ensurer := NewEnsurer(etcdStorage, logger)
+
+			// Call EnsureETCDStatefulSet method and check the result
+			err := ensurer.EnsureETCD(ctx, dummyContext, newEtcd, oldEtcd)
+			Expect(err).To(Not(HaveOccurred()))
+			Expect(*newEtcd.Spec.StorageClass).To(Equal("gardener.cloud-fast"))
+			Expect(*newEtcd.Spec.StorageCapacity).To(Equal(resource.MustParse("80Gi")))
```

Comment:
I would recommend renaming function `checkETCDMain()` to `checkNewETCDMain()` and adding another function `checkOldETCDMain()` which checks for storage capacity to be `80Gi` instead, and use `checkOldETCDMain()` here.

Also, I notice that `checkETCDEvents()` only checks for storage class and not storage capacity. Can you please add a check for storage capacity to be equal to `10Gi` for that?

Reply:
Addressed in [9eab017](https://github.com/gardener/gardener-extension-provider-aws/pull/933/commits/9eab0176574b78dac948e23b12e4c50148b3669c).